File: src/main/java/com/dongfg/project/api/controller/admin/JobController.java
package com.dongfg.project.api.controller.admin;

import com.dongfg.project.api.dto.CommonResponse;
import com.dongfg.project.api.dto.JobInfo;
import com.dongfg.project.api.service.JobService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * @author dongfg
 * @date 18-1-29
 */
@RequestMapping("/admin/task")
@RestController
@Slf4j
public class JobController {
    @Autowired
    private JobService jobService;

    @GetMapping("list")
    public List<JobInfo> list() {
        return jobService.jobInfoList();
    }

    @PostMapping("toggle")
    public CommonResponse<Void> toggle(@RequestParam String group, @RequestParam String name) {
        return jobService.toggleJob(group, name);
    }
}

File: src/main/kotlin/com/dongfg/project/api/web/controller/JobController.kt
package com.dongfg.project.api.web.controller

import com.dongfg.project.api.common.Constants
import com.dongfg.project.api.common.util.EnableSwaggerDoc
import com.dongfg.project.api.common.util.whenInvalid
import com.dongfg.project.api.component.Totp
import com.dongfg.project.api.service.JobService
import com.dongfg.project.api.web.payload.GenericPayload
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@EnableSwaggerDoc
@RequestMapping("/job")
@RestController
class JobController {

    @Autowired
    private lateinit var jobService: JobService

    @Autowired
    private lateinit var totp: Totp

    @GetMapping
    fun jobs(): List<String> {
        return jobService.jobs()
    }

    @GetMapping("/state")
    fun jobState(@RequestHeader("x-auth-totp") totpCode: String, name: String): GenericPayload {
        totp.whenInvalid(totpCode) {
            val payload = GenericPayload(false)
            payload.msg = it
            return payload
        }
        return jobService.jobState(name)
    }

    @PostMapping("/operate")
    fun jobPause(@RequestHeader("x-auth-totp") totpCode: String, name: String, action: Constants.JobAction): GenericPayload {
        totp.whenInvalid(totpCode) {
            val payload = GenericPayload(false)
            payload.msg = it
            return payload
        }
        return jobService.jobOperate(name, action)
    }
}